From 24923e2d437b536b2a48e443a1204ce925e443aa Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Tue, 20 Jun 2023 06:58:41 +0000
Subject: [PATCH] [wasm][cmake] HACK: Tweak standalone stdlib build
 configuration

This commit disables remote mirrors due to cross-compilation issues, and
turns off several additional features that are not yet supported on WASI.
---
 cmake/caches/Runtime-WASI-wasm32.cmake | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/cmake/caches/Runtime-WASI-wasm32.cmake b/cmake/caches/Runtime-WASI-wasm32.cmake
index d96dc885022..2938f4a8f60 100644
--- a/cmake/caches/Runtime-WASI-wasm32.cmake
+++ b/cmake/caches/Runtime-WASI-wasm32.cmake
@@ -8,6 +8,7 @@ set(SWIFT_SDKS WASI CACHE STRING "")
 set(SWIFT_INCLUDE_TOOLS NO CACHE BOOL "")
 set(SWIFT_INCLUDE_TESTS TRUE CACHE BOOL "")
 set(SWIFT_INCLUDE_DOCS NO CACHE BOOL "")
+set(SWIFT_BUILD_REMOTE_MIRROR NO CACHE BOOL "")
 
 set(SWIFT_BUILD_SOURCEKIT NO CACHE BOOL "")
 set(SWIFT_ENABLE_SOURCEKIT_TESTS NO CACHE BOOL "")
@@ -18,8 +19,16 @@ set(SWIFTWASM_DISABLE_REFLECTION_TEST YES CACHE BOOL "")
 set(SWIFT_BUILD_STATIC_STDLIB YES CACHE BOOL "")
 set(SWIFT_BUILD_DYNAMIC_STDLIB NO CACHE BOOL "")
 set(SWIFT_ENABLE_EXPERIMENTAL_CONCURRENCY YES CACHE BOOL "")
+# TODO(katei): This should get turned off, as this is not an ABI stable platform.
+# But current CMake build system doesn't support SWIFT_STDLIB_STABLE_ABI=NO
+set(SWIFT_STDLIB_STABLE_ABI YES CACHE BOOL "")
 
 # build with the host compiler
 set(SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER YES CACHE BOOL "")
 
 set(SWIFT_STDLIB_SINGLE_THREADED_CONCURRENCY TRUE CACHE BOOL "")
+set(SWIFT_THREADING_PACKAGE TRUE CACHE STRING "none")
+
+set(SWIFT_STDLIB_SUPPORTS_BACKTRACE_REPORTING FALSE CACHE BOOL "")
+set(SWIFT_STDLIB_HAS_DLADDR FALSE CACHE BOOL "")
+set(SWIFT_STDLIB_COMPACT_ABSOLUTE_FUNCTION_POINTER TRUE CACHE BOOL "")
-- 
2.40.0

